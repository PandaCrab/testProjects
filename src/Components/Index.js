import { ShippingInfo } from './Forms/ShippingInfo';
import { BillingInfo } from './Forms/BillingInfo';
import { Payment } from './Forms/Payment';
import { OrderPlate } from './OrderPlate/OrderPlate';


export { ShippingInfo, BillingInfo, Payment, OrderPlate };
