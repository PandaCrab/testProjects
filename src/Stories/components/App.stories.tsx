import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ApolloProvider } from '@apollo/client';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from '../../redux/store';
import App from '../../App';
import { apolloClient } from '../..';

export default {
    title: 'App',
    component: App
} as ComponentMeta<typeof App>;

const Template: ComponentStory<typeof App> = () => (
    <ApolloProvider client={apolloClient}>
        <Provider store={store}>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </Provider>
    </ApolloProvider>
);

export const Header = Template.bind({});
