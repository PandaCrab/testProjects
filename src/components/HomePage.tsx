import React from 'react';
import { useNavigate } from 'react-router-dom';

import { 
    StartPage,
    NavigationButton
} from '../styles/HomePageStyles';

const HomePage = () => {
    const navigate = useNavigate();

    return (
        <>
            <StartPage>
                <h1>Hello World</h1>
                <NavigationButton onClick={() => navigate('/shopping')}>Start shopping</NavigationButton>
            </StartPage>
        </>
    );
};

export default HomePage;
