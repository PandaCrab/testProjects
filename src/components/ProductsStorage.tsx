import { useState } from 'react';
import { 
    SortingState,
    IntegratedSorting,
    RowDetailState
} from '@devexpress/dx-react-grid';
import {
  Grid,
  Table,
  TableHeaderRow,
  TableRowDetail,
} from '@devexpress/dx-react-grid-material-ui';
// import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useQuery } from '@apollo/client';

// import { AppDispatch, RootState } from '../types';

// import { getProductsStorage } from '../redux/ducks/stuff';
import { Heading, NavigationButton, StartPage } from '../styles/HomePageStyles';
import { TAKE_PRODUCTS_FROM_STORAGE } from '../api';

interface graphQlData {
    id: string,
    name: string,
    price: number,
    quantity: number
}

const RowDetail = ({ row }: any) => (
    <div>
        {row.quantity > 0 ? 
        `This item can be send today` :
        `Check later, out of stock`
        }
    </div>
)

const ProductsStorage = () => {
    const { error, data } = useQuery(TAKE_PRODUCTS_FROM_STORAGE);
    const navigate = useNavigate();
    // const dispatch = useDispatch<AppDispatch>();
    // const productsStorage = useSelector((state: RootState) => state.order.productsStorage);

    // useEffect(() => { 
    //     dispatch(getProductsStorage())
    // // eslint-disable-next-line react-hooks/exhaustive-deps
    // }, []);

    const [columns] = useState([
        { name: 'id', title: 'Id' },
        { name: 'name', title: 'Name' },
        { name: 'price', title: 'Price' },
        { name: 'quantity', title: 'Quantity' }
    ]);

    const [tableColumnExtensions] = useState([
        { columnName: 'id', width: '10%' },
        { columnName: 'name', width: '50%' },
        { columnName: 'price', width: '10%' },
        { columnName: 'quantity', width: '15%' },
      ]);

    return (
        <StartPage>
            <Heading>
                <h1>Hello in storage</h1>  <NavigationButton onClick={() => navigate('/')} >Home</NavigationButton>
            </Heading>
            <Grid
                rows={data.productsStorage.map(({ id, name, price, quantity }: graphQlData) => ({
                    id,
                    name,
                    price,
                    quantity
                }))}
                columns={columns}
            >
                <SortingState
                    defaultSorting={[{ columnName: 'name', direction: 'asc' }]}
                />
                <IntegratedSorting />
                <RowDetailState />

                <Table columnExtensions={tableColumnExtensions} />
                <TableHeaderRow showSortingControls />
                <TableRowDetail 
                    contentComponent={RowDetail}
                />
            </Grid>
        </StartPage>
    );
};

export default ProductsStorage;
