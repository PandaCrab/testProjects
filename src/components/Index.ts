import ShippingInfo from './forms/ShippingInfo';
import BillingInfo from './forms/BillingInfo';
import Payment from './forms/Payment';
import OrderPlate from './orderPlate/OrderPlate';
import SuccessBuying from './forms/SuccessBuying';
import Navigate from '../Navigate';
import Stuff from './orderPlate/Stuff';
import StuffItems from './orderPlate/StuffItems';
import Loader from './Loader';
import CardToPrint from './forms/CardToPrint';
import DropdownAddresses from './forms/dropdownAddresses';
import CountriesSelect from './forms/CountriesSelect';
import HomePage from './HomePage';
import ProductsStorage from './ProductsStorage';
import SaleChart from './SaleChart';

export { ShippingInfo,
    BillingInfo, 
    Payment, 
    OrderPlate, 
    SuccessBuying, 
    Navigate,
    Stuff,
    StuffItems,
    Loader,
    CardToPrint,
    DropdownAddresses,
    CountriesSelect,
    HomePage,
    ProductsStorage,
    SaleChart
};
