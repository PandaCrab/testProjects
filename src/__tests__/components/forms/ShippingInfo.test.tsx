import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { ShippingInfo } from '../../../components/Index';

describe('shipping info component:', () => {
    const initialState: any = {address: {
        navigatorAddress: []
    }};
    const mockStore = configureStore();
    let store;
    
    it('should render correctly Shipping info component', () => {
        store = mockStore(initialState);
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <ShippingInfo />
                </BrowserRouter>
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

});
