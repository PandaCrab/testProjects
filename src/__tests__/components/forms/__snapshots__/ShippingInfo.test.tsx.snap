// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shipping info component: should render correctly Shipping info component 1`] = `
Array [
  <section
    className="sc-cOFTSb foYTCl"
  >
    <a
      href="/shipping"
      onClick={[Function]}
    >
      Shipping
    </a>
     &gt;
    <a
      href="/billing"
      onClick={[Function]}
    >
      Billing
    </a>
     &gt;
    <a
      href="/payment"
      onClick={[Function]}
    >
      Payment
    </a>
  </section>,
  <div
    className="sc-bczRLJ iLjvJl"
  >
    <section
      className="sc-jqUVSM drOJkA"
    >
      <h4
        className="sc-kDDrLX hxhBIM"
      >
        Shipping Information
      </h4>
    </section>
    <form
      className=""
      noValidate={true}
      onSubmit={[Function]}
    >
      <div>
        <div>
          <label
            className="mb-0 form-label"
          >
            Recipient
          </label>
          <div
            className="mb-4 position-relative"
          >
            <input
              className="form-control"
              id="Name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Full Name"
              type="text"
              value=""
            />
            <div
              className="invalid-tooltip"
            />
          </div>
          <div
            className="mb-4 mb-md-4 mb-lg-5 mb-xl-5 position-relative"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-7 col-7"
              >
                <div
                  className=" react-tel-input sc-jSMfEi hECgnK form-control"
                  onKeyDown={[Function]}
                  style={Object {}}
                >
                  
                  <input
                    className="form-control "
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onCopy={[Function]}
                    onDoubleClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Daytime Phone"
                    style={Object {}}
                    type="tel"
                    value=""
                  />
                  <div
                    className="flag-dropdown "
                    style={Object {}}
                  >
                    <div
                      aria-haspopup="listbox"
                      className="selected-flag"
                      onClick={[Function]}
                      role="button"
                      tabIndex="0"
                      title=""
                    >
                      <div
                        className="flag 0"
                      >
                        <div
                          className="arrow"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="invalid-tooltip"
                />
              </div>
              <div
                className="col-sm-4 col-5"
              >
                <p
                  className="sc-ftvSup dgAAPZ"
                >
                  For delivery questions only
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="mb-4 mb-md-3 mb-lg-4 mb-xl-4 position-relative"
        >
          <label
            className="mb-0 form-label"
          >
            Address
          </label>
          <div
            className="mb-4 mb-md-3 mb-lg-4 mb-xl-4 position-relative"
          >
            <input
              className="form-control"
              id="street"
              name="street"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Street address"
              type="text"
              value=""
            />
            <svg
              className="sc-dkzDqf fVHUGe"
              fill="currentColor"
              height="1em"
              onClick={[Function]}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 16s6-5.686 6-10A6 6 0 0 0 2 6c0 4.314 6 10 6 10zm0-7a3 3 0 1 1 0-6 3 3 0 0 1 0 6z"
              />
            </svg>
            <div
              className="invalid-tooltip"
            />
          </div>
          <div
            className="mb-2 mb-md-3 mb-lg-4 mb-xl-4"
          >
            <input
              className="form-control"
              id="optional"
              name="optional"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Apt, Suit, Bidg, Gate Code. (optional)"
              type="text"
              value=""
            />
          </div>
          <div
            className="mb-4 mb-md-3 mb-lg-4 mb-xl-4 position-relative"
          >
            <input
              className="form-control"
              id="City"
              name="city"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="City"
              type="text"
              value=""
            />
            <div
              className="invalid-tooltip"
            />
          </div>
          <div
            className="mb-2 mb-md-3 mb-lg-4 mb-xl-4"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-7 col-7"
              >
                <div
                  className=" css-b62m3t-container"
                  id="country"
                  onKeyDown={[Function]}
                >
                  <span
                    className="css-1f43avz-a11yText-A11yText"
                    id="react-select-2-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    className="css-1f43avz-a11yText-A11yText"
                  />
                  <div
                    className=" css-1s2u09g-control"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <div
                      className=" css-319lph-ValueContainer"
                    >
                      <div
                        className=" css-14el2xx-placeholder"
                        id="react-select-2-placeholder"
                      >
                        Country
                      </div>
                      <div
                        className=" css-6j8wv5-Input"
                        data-value=""
                      >
                        <input
                          aria-autocomplete="list"
                          aria-describedby="react-select-2-placeholder"
                          aria-expanded={false}
                          aria-haspopup={true}
                          autoCapitalize="none"
                          autoComplete="off"
                          autoCorrect="off"
                          className=""
                          disabled={false}
                          id="react-select-2-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          role="combobox"
                          spellCheck="false"
                          style={
                            Object {
                              "background": 0,
                              "border": 0,
                              "color": "inherit",
                              "font": "inherit",
                              "gridArea": "1 / 2",
                              "label": "input",
                              "margin": 0,
                              "minWidth": "2px",
                              "opacity": 1,
                              "outline": 0,
                              "padding": 0,
                              "width": "100%",
                            }
                          }
                          tabIndex={0}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      className=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        className=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        className=" css-tlfecz-indicatorContainer"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <svg
                          aria-hidden="true"
                          className="css-tj5bde-Svg"
                          focusable="false"
                          height={20}
                          viewBox="0 0 20 20"
                          width={20}
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="country"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
              <div
                className="col-sm-5 col-5"
              >
                <input
                  className="form-control"
                  id="Zip"
                  name="zip"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="ZIP"
                  type="text"
                  value=""
                />
                <div
                  className="invalid-tooltip"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="sc-papXJ fRWQUH btn btn-primary"
        disabled={false}
        type="submit"
      >
        Continue
      </button>
    </form>
  </div>,
]
`;
